ambry-tools/src/main/java/com.github.ambry.tools/admin/DumpData.java

```diff
@@ -464,384 +451,335 @@ public long dumpActiveBlobsFromIndex(File indexFileToDump, ArrayList<String> blo
       throws Exception {
     if (logFileToDump == null) {
-      System.out.println("logFileToDump needs to be set for compareIndexToLog");
+      log("logFileToDump needs to be set for compareIndexToLog");
       System.exit(0);
     }
-    DataInputStream stream = new DataInputStream(new FileInputStream(logFileToDump));
-    RandomAccessFile randomAccessFile = new RandomAccessFile(new File(logFileToDump), "r");
-    System.out.println("Comparing Index entries to Log ");
-    short version = stream.readShort();
-    System.out.println("version " + version);
-    if (version == 0) {
-      int keysize = stream.readInt();
-      int valueSize = stream.readInt();
-      long fileEndPointer = stream.readLong();
-      System.out.println("key size " + keysize);
-      System.out.println("value size " + valueSize);
-      System.out.println("file end pointer " + fileEndPointer);
-      int Crc_Size = 8;
-      StoreKeyFactory storeKeyFactory = Utils.getObj("com.github.ambry.commons.BlobIdFactory", map);
-      while (stream.available() > Crc_Size) {
-        StoreKey key = storeKeyFactory.getStoreKey(stream);
-        byte[] value = new byte[IndexValue.Index_Value_Size_In_Bytes];
-        stream.read(value);
-        IndexValue blobValue = new IndexValue(ByteBuffer.wrap(value));
-        String msg = "key :" + key + ": value - offset " + blobValue.getOffset() + " size " +
-            blobValue.getSize() + " Original Message Offset " + blobValue.getOriginalMessageOffset() +
-            " Flag " + blobValue.getFlags() + "\n";
-        boolean success = readFromLog(randomAccessFile, blobValue.getOffset(), map, key.getID());
-        if (!success) {
-          System.out.println("Failed for Index Entry " + msg);
-        }
+    RandomAccessFile randomAccessFile = null;
+    try {
+      randomAccessFile = new RandomAccessFile(new File(logFileToDump), "r");
+      log("Comparing Index entries to Log ");
+      compareIndexEntryToLogContent(new File(indexFile), null, randomAccessFile, avoidTraceLogging);
+    } finally {
+      if (randomAccessFile != null) {
+        randomAccessFile.close();
       }
-      System.out.println("crc " + stream.readLong());
     }
   }
 
-  public boolean readFromLog(RandomAccessFile randomAccessFile, long offset, ClusterMap map, String blobId)
+  /**
+   * Compares every entry in an index file with those in the log. Checks to see if each blob in index is successfully deserializable
+   * from the log
+   * @param indexFile the index file that needs to be checked for
+   * @param replicaDirectory the replica root directory where the index is located
+   * @param randomAccessFile the {@link RandomAccessFile} referring to the log file
+   * @param avoidTraceLogging {@code true} if miscellaneous logging need to be avoided, {@code false} otherwise
+   * @throws Exception
+   */
+  private void compareIndexEntryToLogContent(File indexFile, File replicaDirectory, RandomAccessFile randomAccessFile,
```

Comment:
Is it necessary to have both this method and `compareIndexEntriestoLogContent`. The two names are also a bit confusing.